hoomd/hpmc/IntegratorHPMCMono.h

```diff
@@ -333,17 +333,24 @@ class IntegratorHPMCMono : public IntegratorHPMC
             // base class method
             IntegratorHPMC::prepRun(timestep);
 
+            m_hasOrientation = false;
+            quat<Scalar> q(make_scalar4(1,0,0,0));
+            for (unsigned int i=0; i < m_pdata->getNTypes(); i++)
                 {
-                // for p in params, if Shape dummy(q_dummy, params).hasOrientation() then m_hasOrientation=true
-                m_hasOrientation = false;
-                quat<Scalar> q(make_scalar4(1,0,0,0));
-                for (unsigned int i=0; i < m_pdata->getNTypes(); i++)
+                Shape dummy(q, m_params[i]);
+                if (dummy.hasOrientation())
                     {
-                    Shape dummy(q, m_params[i]);
-                    if (dummy.hasOrientation())
-                        m_hasOrientation = true;
+                    m_hasOrientation = true;
+                    break;
                     }
                 }
+
+            //! Update the OBB tree if necessary
+            if (m_patch && m_patch->getBuildOBB())
+                {
+                m_patch->buildOBBTree();
```

Comment:
I'm not familiar enough with the patch code to know what this does, but it seems odd to me that a user parameter can influence whether or not the tree is built. I would think that the tree build always needs to be called and the callee can decide whether the call does anything or not.

Reply:
I believe that is how it was done in the original implementation where the tree was built inside the `setParams` method, so every `patch.set_params(...)` call by the user would trigger the tree build. I can certainly move the build flag check into the `buildOBBTree`method and call it every time within the integrator as is done with the AABB tree.